File: ChatGPT/Direct_Attempt/python/output/output_25a.py
# ChatGPT-generated solution will go here.
def parse_input():
    with open("input.txt") as f:
        blocks = f.read().strip().split("\n\n")

    def to_heights(block, is_lock):
        lines = block.splitlines()
        W = len(lines[0])
        H = len(lines)
        heights = []
        for col in range(W):
            h = 0
            if is_lock:
                for row in range(H):
                    if lines[row][col] == "#":
                        h += 1
                    else:
                        break
            else:
                for row in range(H - 1, -1, -1):
                    if lines[row][col] == "#":
                        h += 1
                    else:
                        break
            heights.append(h)
        return heights

    locks, keys = [], []
    for block in blocks:
        lines = block.strip().splitlines()
        if lines[0].startswith("#"):
            locks.append(to_heights(block, True))
        else:
            keys.append(to_heights(block, False))
    return locks, keys

def count_fitting_pairs(locks, keys):
    count = 0
    for l in locks:
        for k in keys:
            if all(l[i] + k[i] <= len(l) for i in range(len(l))):
                count += 1
    return count

def main():
    locks, keys = parse_input()
    print(count_fitting_pairs(locks, keys))

if __name__ == "__main__":
    main()

File: ChatGPT/Direct_Attempt/clojure/output/output_25a.clj
;; ChatGPT-generated solution will go here.
(ns output_25a
  (:require [clojure.string :as str]))

(defn parse-schematics [lines]
  (let [groups (->> lines
                    (partition-by str/blank?)
                    (remove #(= '("") %)))]
    (map #(apply map vector %) groups))) ; transpose to get columns

(defn schematic-height [row-selector grid]
  (map (fn [col]
         (count (take-while row-selector col)))
       grid))

(defn lock-heights [grid]
  (schematic-height #(= \# %) grid))

(defn key-heights [grid]
  (schematic-height #(= \# %) (map reverse grid)))

(defn part [lines]
  (let [schematics (parse-schematics lines)
        [locks keys] (split-with #(= \# (ffirst %)) schematics)
        locks (map lock-heights locks)
        keys (map key-heights keys)
        total-height (count (first (first schematics)))
        valid-pairs (for [l locks
                          k keys
                          :when (every? #(< % total-height) (map + l k))]
                      [l k])]
    (count valid-pairs)))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
